src/Microsoft.DotNet.Cli.Utils/CommandResolution/AppBaseCommandResolver.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class AppBaseCommandResolver : ICommandResolver
+    {
+        public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
+        {
+            if (commandResolverArguments.CommandName == null)
+            {
+                return null;
+            }
+
+            var commandPath = commandResolverArguments.Environment.GetCommandPathFromRootPath(
+                PlatformServices.Default.Application.ApplicationBasePath, 
+                commandResolverArguments.CommandName);
+
+            return commandPath == null
+                ? null
+                : CommandResolverCommon.CreateCommandSpecPreferringExe(
```

Comment:
Why not ICommandSpecFactory then? And then have an implementation for it called ExePrefferedCommandSpecFactory.